pkg/client/cache/index_test.go

```diff
@@ -48,3 +50,46 @@ func TestGetIndexFuncValues(t *testing.T) {
 		}
 	}
 }
+
+func testUsersIndexFunc(obj interface{}) ([]string, error) {
+	pod := obj.(*api.Pod)
+	usersString := pod.Annotations["users"]
+
+	return strings.Split(usersString, ","), nil
+}
+
+func TestMultiIndexKeys(t *testing.T) {
```

Comment:
Can you add a little more to this test scenario?  For example, modify the pod, and replace it in index and ensure that it is not retrievable by older index keys?